plugin/pkg/admission/controller/admission.go

```diff
@@ -0,0 +1,81 @@
+package controller
+
+import (
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/admission"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/client/cache"
+	client "k8s.io/kubernetes/pkg/client/unversioned"
+	controller "k8s.io/kubernetes/pkg/controller/replication"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/labels"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch"
+
+)
+
+var (
+	selectorKeyFunc = controller.NamespaceSelectorKeyFunc
+)
+
+
+func init() {
+	admission.RegisterPlugin("ReplicationControllerExists", func(client client.Interface, config io.Reader) (admission.Interface, error) {
+		return NewExists(client), nil
+	})
+}
+
+// exists is an implementation of admission.Interface.
+// It rejects all incoming CREATE or UPDATE requests of replication controller if it exists
+// It is useful in deployments that want to filter out repeated replication controller
+type exists struct {
+	*admission.Handler
+	client client.Interface
+	store  cache.Store
+}
+
+func (e *exists) Admit(a admission.Attributes) (err error) {
+	if a.GetResource() == "replicationcontrollers" {
+		_, ok := a.GetObject().(*api.ReplicationController)
+		if !ok {
+			return nil
+		}
+		key, err := selectorKeyFunc(a.GetObject())
```

Comment:
Doesn't this result in an error any time you edit an existing ReplicationController?